File: 双指针 two pointers/0027. 移除元素/移除元素.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        if len(nums)==0:
            return 0
        p=0
        q=0
        while q<len(nums):
            if nums[q]==val:
                q=q+1
            else:
                nums[p]=nums[q]
                q=q+1
                p=p+1
        return p

File: 双指针 two pointers/0027. 移除元素/移除元素.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        if(nums.size()==0)
            return 0;
        int p=0,q=0;
        while(q<nums.size())
        {
            if(nums[q]==val)
            {
                q++;
            }
            else
            {
                nums[p]=nums[q];
                p++;
                q++;
            }
        }
        return p;
    }
};